packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -430,12 +430,6 @@ class IOSSimulator extends Device {
         return new LaunchResult.failed();
     }
 
-    ProtocolDiscovery observatoryDiscovery;
-
-    if (debuggingOptions.debuggingEnabled)
-      observatoryDiscovery = new ProtocolDiscovery(getLogReader(app: app),
```

Comment:
Moving this below the `SimControl.instance.launch()` line introduces a race condition whereby we may not start tailing the logs until after the observatory port log line has already passed.